### DISCUSSION TITLE
`getServerSideProps` always returning 404 when deployed on Vercel

### DISCUSSION THREAD

**Discussion by manojahi** [vercel/community#1849] [Help]
### Summary

I have recently started with NextJs and like it so far.

I have an index page where I am having server-side rendering. I have used getServerSideProps to do this.

I have the following code in my index.js

```js
import CardGrid from "../components/cardGrid";
import ErrorPage from "next/error";

function Home({
  wallData = []
}) {
  return (
    <>
      <div className="flex flex-col h-screen">
        <div className="container mx-auto flex-grow">
          <CardGrid
            wallData={wallData}
          />
        </div>
      </div>
    </>
  );
}

export async function getServerSideProps(context) {
  return {
    props: {
    },
  };
}

export default Home;
```

It's working fine when running locally but when I deployed it to Vercel, it always gives a 404 error.

I removed the use of getServerSideProps and deployed it, that time it worked on Vercel but with getServerSideProps, it just gave a 404 error.

Tried to check the Vercel logs but it's having no information at all.

```
MAR 21 23:39:17.29
404
xxxxxxxx.vercel.app
[GET] /
```

I tried to search a lot but could not get it to work, would appreciate any help.

**Comment by amyegan** [comment]
Hi @manojahi 👋 I tried using the code you shared to reproduce the problem in my test app, but I'm not getting the error. I suspect there is an interaction between the page you shared and some other setting in your app. Are you able to share the repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can get a better look at the problem?

**Reply by manojahi** [reply]
Issue is fixed now. There was some problem with my `next.config.js`. 
thanks for your help.

**Reply by JacoboArcila** [reply]
Hello, I have the same error. How did you fix it?
